frontend/lib/src/dataframes/Quiver.ts

```diff
@@ -914,15 +651,18 @@ st.add_rows(my_styler.data)
       draft._types = types
     })
   }
+}
 
-  private static parseFields(schema: ArrowSchema): Record<string, Field> {
-    // None-index data columns are listed first, and all index columns listed last
-    // within the fields array in arrow.
-    return Object.fromEntries(
-      (schema.fields || []).map((field, index) => [
-        field.name.startsWith("__index_level_") ? field.name : String(index),
-        field,
-      ])
-    )
+/** Parse styler information from proto. */
+function parseStyler(styler: StylerProto): Styler {
```

Comment:
Unfortunately, we cannot easily move this method to `arrowParseUtils` since it would cause a cyclic import. But I will look into more refactoring around the styler aspects in subsequent PRs.